### ISSUE TITLE
results (data) are not fetched with `datalad-pair` orc on localshell

### ISSUE THREAD

**Issue by yarikoptic** [ReproNim/reproman#462]
```shell
hopa:/tmp
$> sudo rm -rf testds; ( datalad create testds; cd testds; reproman run --follow -r localshell --sub condor --orc datalad-pair bash -c 'which datalad >| .dotfile;  datalad --version >| file'; git status; ls -l .dotfile file; grep . .dotfile file; ) 
[INFO   ] Creating a new annex repo at /tmp/testds 
create(ok): /tmp/testds (dataset)                                                                                                                                            
2019-09-26 13:09:48,759 [INFO   ] No root directory supplied for localshell; using '/home/yoh/.reproman/run-root' 
[INFO   ] Cloning /tmp/testds into '/home/yoh/.reproman/run-root/70f8bcd8-e080-11e9-bde9-8019340ce7f2' 
2019-09-26 13:09:49,545 [INFO   ] Submitting 20190926-130947-8b81 
2019-09-26 13:09:49,578 [INFO   ] Submitting /home/yoh/.reproman/run-root/70f8bcd8-e080-11e9-bde9-8019340ce7f2/.reproman/jobs/localshell/20190926-130947-8b81/submit 
2019-09-26 13:09:49,634 [INFO   ] Job 20190926-130947-8b81 submitted as condor job 26 
2019-09-26 13:09:49,643 [INFO   ] Registered job 20190926-130947-8b81 
2019-09-26 13:09:49,660 [INFO   ] Waiting on job 26: running 
2019-09-26 13:09:59,732 [INFO   ] Fetching results for 20190926-130947-8b81 
2019-09-26 13:09:59,869 [INFO   ] Unregistered job 20190926-130947-8b81 
On branch master
nothing to commit, working tree clean
lrwxrwxrwx 1 yoh yoh 110 Sep 26 13:09 .dotfile -> .git/annex/objects/MP/X3/MD5E-s60--804239fecdbf5c243f91122966d44007/MD5E-s60--804239fecdbf5c243f91122966d44007
lrwxrwxrwx 1 yoh yoh 110 Sep 26 13:09 file -> .git/annex/objects/jG/71/MD5E-s25--64271e240e83266c1bcb991f0999ffec/MD5E-s25--64271e240e83266c1bcb991f0999ffec
grep: .dotfile: No such file or directory
grep: file: No such file or directory

$> grep .  /home/yoh/.reproman/run-root/70f8bcd8-e080-11e9-bde9-8019340ce7f2/{.dotfile,file}
/home/yoh/.reproman/run-root/70f8bcd8-e080-11e9-bde9-8019340ce7f2/.dotfile:/home/yoh/proj/repronim/reproman-run/venvs/dev3/bin/datalad
/home/yoh/.reproman/run-root/70f8bcd8-e080-11e9-bde9-8019340ce7f2/file:datalad 0.12.0rc5.dev102

$> reproman --version
reproman 0.2.0
```

<details>
<summary>I thought that may be it is because `--output`s were not explicitly specified, but adding `--output` s didn't help:</summary>

```shell
$> sudo rm -rf testds; ( datalad create testds; cd testds; reproman run --output '*' --output '.dot*' --follow -r localshell --sub condor --orc datalad-pair bash -c 'which datalad >| .dotfile;  datalad --version >| file'; git status; ls -l .dotfile file; grep . .dotfile file; )
[INFO   ] Creating a new annex repo at /tmp/testds 
create(ok): /tmp/testds (dataset)                                                                                                                                            
2019-09-26 13:12:54,848 [INFO   ] No root directory supplied for localshell; using '/home/yoh/.reproman/run-root' 
[INFO   ] Cloning /tmp/testds into '/home/yoh/.reproman/run-root/dff1adde-e080-11e9-bde9-8019340ce7f2' 
2019-09-26 13:12:55,642 [INFO   ] Submitting 20190926-131253-19b7 
2019-09-26 13:12:55,674 [INFO   ] Submitting /home/yoh/.reproman/run-root/dff1adde-e080-11e9-bde9-8019340ce7f2/.reproman/jobs/localshell/20190926-131253-19b7/submit 
2019-09-26 13:12:55,733 [INFO   ] Job 20190926-131253-19b7 submitted as condor job 27 
2019-09-26 13:12:55,741 [INFO   ] Registered job 20190926-131253-19b7 
2019-09-26 13:12:55,757 [INFO   ] Waiting on job 27: idle 
2019-09-26 13:13:05,808 [INFO   ] Waiting on job 27: idle 
2019-09-26 13:13:15,879 [INFO   ] Fetching results for 20190926-131253-19b7 
2019-09-26 13:13:16,028 [INFO   ] Unregistered job 20190926-131253-19b7 
On branch master
nothing to commit, working tree clean
lrwxrwxrwx 1 yoh yoh 110 Sep 26 13:13 .dotfile -> .git/annex/objects/MP/X3/MD5E-s60--804239fecdbf5c243f91122966d44007/MD5E-s60--804239fecdbf5c243f91122966d44007
lrwxrwxrwx 1 yoh yoh 110 Sep 26 13:13 file -> .git/annex/objects/jG/71/MD5E-s25--64271e240e83266c1bcb991f0999ffec/MD5E-s25--64271e240e83266c1bcb991f0999ffec
grep: .dotfile: No such file or directory
grep: file: No such file or directory
```
</details>

Then I thought that may be I have to fetch explicitly using `reproman jobs`, but the job was unregistered and there is no remote known within testds

**Comment by yarikoptic** [comment]
<details>
<summary>
FTR: works as I expect it with `datalad-pair-run` orchestrator:
</summary> 

```shell
$> sudo rm -rf testds; ( datalad create testds; cd testds; reproman run --follow -r localshell --sub condor --orc datalad-pair-run bash -c 'which datalad >| .dotfile;  datalad --version >| file'; git status; ls -l .dotfile file; grep . .dotfile file; ) 
[INFO   ] Creating a new annex repo at /tmp/testds                                                                                 
create(ok): /tmp/testds (dataset)                                                                                                                                            
2019-09-26 13:20:29,417 [INFO   ] No root directory supplied for localshell; using '/home/yoh/.reproman/run-root' 
[INFO   ] Cloning /tmp/testds into '/home/yoh/.reproman/run-root/eebec0b2-e081-11e9-bde9-8019340ce7f2' 
2019-09-26 13:20:30,272 [INFO   ] Submitting 20190926-132028-5885 
2019-09-26 13:20:30,320 [INFO   ] Submitting /home/yoh/.reproman/run-root/eebec0b2-e081-11e9-bde9-8019340ce7f2/.reproman/jobs/localshell/20190926-132028-5885/submit 
2019-09-26 13:20:30,379 [INFO   ] Job 20190926-132028-5885 submitted as condor job 29 
2019-09-26 13:20:30,388 [INFO   ] Registered job 20190926-132028-5885 
2019-09-26 13:20:30,406 [INFO   ] Waiting on job 29: running 
2019-09-26 13:20:40,474 [INFO   ] Fetching results for 20190926-132028-5885 
2019-09-26 13:20:40,544 [INFO   ] Creating run commit in /tmp/testds 
2019-09-26 13:20:40,982 [INFO   ] Unregistered job 20190926-132028-5885                                                                                                      
On branch master
nothing to commit, working tree clean
lrwxrwxrwx 1 yoh yoh 110 Sep 26 13:20 .dotfile -> .git/annex/objects/MP/X3/MD5E-s60--804239fecdbf5c243f91122966d44007/MD5E-s60--804239fecdbf5c243f91122966d44007
lrwxrwxrwx 1 yoh yoh 110 Sep 26 13:20 file -> .git/annex/objects/jG/71/MD5E-s25--64271e240e83266c1bcb991f0999ffec/MD5E-s25--64271e240e83266c1bcb991f0999ffec
.dotfile:/home/yoh/proj/repronim/reproman-run/venvs/dev3/bin/datalad
file:datalad 0.12.0rc5.dev102
```
</details>

**Comment by kyleam** [comment]
tl;dr  Right, the pair/local combo doesn't result in content getting fetched.  The current code was intended to be a stopgap for testing purposes until `datalad create-sibling` supports working with local paths.

---

For the `datalad-pair` orchestrator, the local shell code paths for both preparing the remote and fetching the results are separate from ssh code paths.  Here's the fetching bit:

https://github.com/ReproNim/reproman/blob/62084cb4eb404d05895e1b189ff69566afb10344/reproman/support/jobs/orchestrators.py#L936-L938

As mentioned by that comment, the current code exists for testing purposes so that at least something works for a local shell resource, but it doesn't bother with getting the content.  My current thinking, which is touched upon in a good number of commit messages (e.g., 0ed4551, 321a407a9, a028908, and 6ce323f), is that we should address this by adding local path support to `create-sibling` and then sending local resources down the same code path.
